backend/pkg/models/v1beta/job_offers.go

```diff
@@ -68,6 +78,51 @@ func (r OfferPostQuery) Validate() error {
 		}
 	}
 
+	if strings.TrimSpace(r.ApplyPhoneNumber) != "" {
+		phoneRegex := regexp.MustCompile(`^[+]*[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s\./0-9]*$`)
+		if !phoneRegex.MatchString(r.ApplyPhoneNumber) {
+			return errors.New("apply phone number is not a valid phone number")
+		}
+	}
```

Comment:
Maybe we could extract all the regex matches we are doing here in a separate function that can have unit tests.
The idea will be to show what we are expecting as input for this regex "matcher".